From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Wed, 15 Jul 2026 20:57:42 +0200
Subject: arm64: dts: allwinner: sun50i-a64: Change A64 SoC usb otg compatible

The controllers are not really compatible, when it comes to DMA
implementation.

The USB-OTG internal DMA registers live at 0x1c19500, above the USB PHY
and outside the 0x400 the node declares, so the driver was reaching them
through a mapping it was never granted. Describe them as a second
region.

The three windows now tile without overlapping:

  0x1c19000 +0x400  musb core
  0x1c19400 +0x100  usbphy phy_ctrl
  0x1c19500 +0x100  musb idma
Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64.dtsi | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-a64.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64.dtsi
@@ -631,8 +631,11 @@ msgbox: mailbox@1c17000 {
 		};
 
 		usb_otg: usb@1c19000 {
-			compatible = "allwinner,sun8i-a33-musb";
-			reg = <0x01c19000 0x0400>;
+			compatible = "allwinner,sun50i-a64-musb",
+				     "allwinner,sun8i-a33-musb";
+			reg = <0x01c19000 0x0400>,
+			      <0x01c19500 0x0100>;
+			reg-names = "mc", "dma";
 			clocks = <&ccu CLK_BUS_OTG>;
 			resets = <&ccu RST_BUS_OTG>;
 			interrupts = <GIC_SPI 71 IRQ_TYPE_LEVEL_HIGH>;
-- 
Armbian

